egs/wsj/s5/steps/durmod/durmodel_run.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Copyright 2015 Hossein Hadian
+
+# This script calls all the duration modeling scripts with default settings and does the
+# rescoring.
+
+testdata=
+cmd=run.pl
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+if [ $# != 4 ]; then
+   echo "Usage: $0 [options] <phones-dir> <ali-dir> <decode-dir> <rescored-decode-dir>"
+   echo "e.g.: $0 data/lang/phones exp/mono_ali exp/mono/decode_test_bg exp/mono/decode_test_bg_durmod"     
+   exit 1;
+fi
+
+phones_dir=$1
+alidir=$2
+decode_dir=$3
+dir=$4
+
+steps/durmod/durmodel_prepare_examples.sh --left-context 4 \
```

Comment:
Wouldn't it make more sense to have durmodel_prepare_examples.sh called from durmodel_train.sh?  This is the normal pattern.  You could make it one stage of the script [be sure to to support the --stage option].

In addition, this script, steps/durmod/durmodel_run.sh, probably shouldn't exist.  You should be making separate calls to the training and the rescoring scripts, in some local script, e.g. [in some particular setup] local/run_durmodel.sh.